app.py

```diff
@@ -36,52 +46,142 @@
     page_icon="📦",
     initial_sidebar_state="expanded",
 )
-ONE_DAY = 60 * 60 * 24
-TWO_HOURS = 60 * 60 * 4
-THIRTY_MINS = 60 * 30
-FIFTEEN_MINS = 60 * 15
-CHANNELS = ["conda-forge", "bioconda", "pkgs/main", "pkgs/r"]
-# pkgs/msys2 does not seem to offer .conda artifacts; leave out for now
-
 
 def bar_esc(s):
     "Escape vertical bars in tables"
     return s.replace("|", "\\|")
 
 
-@st.cache_resource(ttl=FIFTEEN_MINS, max_entries=5)
-def rssdata(channel="conda-forge"):
-    if channel.startswith("pkgs/"):
-        r = requests.get(f"https://repo.anaconda.com/{channel}/rss.xml")
-    else:
-        r = requests.get(f"https://conda.anaconda.org/{channel}/rss.xml")
+@st.cache_resource
+def app_config() -> AppConfig:
+    return AppConfig()
+
+
+def get_channel_config(channel_name: str) -> Channel:
+    try:
+        return app_config().channels[channel_name]
+    except KeyError:
+        raise ValueError(f"Channel `{channel_name}` not found in the configuration!")
+
+
+def _unwrap_secret(secret: str | Secret) -> str:
+    if isinstance(secret, str):
+        return secret
+    return secret.get_value()
+
+
+def _make_http_session(channel_name: str) -> requests.Session:
+    session = requests.Session()
+    session.headers["User-Agent"] = "conda-metadata-browser/0.1.0"
+    channel_config = get_channel_config(channel_name)
+    if channel_config.auth_username is not None:
+        # use HTTP basic auth
+        username = _unwrap_secret(channel_config.auth_username)
+        password = _unwrap_secret(channel_config.auth_password)
+        session.auth = HTTPBasicAuth(username, password)
+        return session
+    if quetz_token := channel_config.auth_quetz_token:
+        # use Quetz token auth
+        session.headers["X-API-Key"] = _unwrap_secret(quetz_token)
+        return session
```

Comment:
Im assuming private anaconda.org channels don't support this, right? Since adding the whole /t/<token> to the url, I think as well it's okay to not support this for now but maybe in the future if there is need

Reply:
Yes, exactly.